### PR TITLE
Turned exercise 059 into practice

### PR DESCRIPTION
**Pull request by maha-sachin**
Turned exercise 059 into practice 059

Added test cases for height-balanced binary trees

- Implemented additional test cases to ensure correctness and coverage for the height-balanced binary tree generation function.
- Test cases cover various scenarios including empty trees, single-node trees, trees with different heights, asymmetric height trees, large height trees, and random trees.
- Each test case verifies the expected height of the generated tree.
- Ensured that the test suite comprehensively checks the functionality of the hbal_tree function and the tree_height helper function.

### CODE REVIEW

File: practice/059/ex.ml
Comment (cuihtlauac) [inline practice/059/ex.ml:1]: Add more tests. Possibly using trees from other exercises

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/059/ex.ml
Comment (maha-sachin) [reply practice/059/ex.ml:1]: Hi @SaySayo & @cuihtlauac, I'm encountering some errors while running tests for the Height-Balanced Binary Trees module in OCaml. When I run the dune test --root . answer command, I'm seeing the following error messages:
Could you please review the error messages and provide guidance on how to troubleshoot and fix the problem? I'm particularly confused about why the tests for Test Height 2 Trees and Test Height 3 Trees are failing.

Any help or suggestions you could provide would be greatly appreciated.

Comment (cuihtlauac) [reply practice/059/ex.ml:1]: Please commit your changes to the PR, I can't reproduce the issue you are facing

Comment (maha-sachin) [reply practice/059/ex.ml:1]: > Please commit your changes to the PR, I can't reproduce the issue you are facing

okay

Comment (maha-sachin) [reply practice/059/ex.ml:1]: > > Please commit your changes to the PR, I can't reproduce the issue you are facing
> 


@cuihtlauac, Sure, I've committed the changes to the pull request. Please review them when you have a moment.

Comment (cuihtlauac) [reply practice/059/ex.ml:1]: This is your test `"Test Height 3 Tree"` which is failing. `expected_trees` is not equal to `hbal_tree 3`

Comment (maha-sachin) [reply practice/059/ex.ml:1]: yes

**Review by cuihtlauac — APPROVED** [review]
Thanks @maha-sachin, this is well done

**Comment by maha-sachin** [comment]
> Thanks @maha-sachin, this is well done

@cuihtlauac, Thank you for your guidance and support.

**Pull request merged**